src/components/VetoDialog/VetoDialog.tsx

```diff
@@ -0,0 +1,98 @@
+import { AddressBadge, Button, ButtonStyle, Dialog, DialogBody, DialogHeader } from "@sovryn/ui";
+import { useRefetchTrigger, useTxStore } from "../../lib/tx-store";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { Contract, constants } from "ethers";
+import { MULTISIG_CONTRACT_ADDRESS } from "../../config/constants";
+import { getProvider } from "@sovryn/ethers-provider";
+import { CHAIN_ID } from "../../config/network";
+import abi from '../../abi/multisig.json';
+import { useAccount } from "../../hooks/useAccount";
+import { generateCancelData } from "../../lib/helpers";
+import { LinkHashToExplorer } from "../LinkToExplorer/LinkToExplorer";
+
+export const VetoDialog = () => {
+  const msContract = useRef(new Contract(MULTISIG_CONTRACT_ADDRESS, abi, getProvider(CHAIN_ID)));
```

Comment:
```suggestion
  const contract = useRef(new Contract(MULTISIG_CONTRACT_ADDRESS, abi, getProvider(CHAIN_ID)));
```
This can be defined above the component, you don't need `ref` for it.